scripts/run-on-vm.sh

```
@@ -14,13 +15,25 @@ while [[ "${1:-}" == --* ]]; do
         --load)    RUN_PHASE="load"; TARBALL="$2"; shift 2 ;;
+        --keep-tmp) KEEP_TMP=true; shift ;;
         *) echo "Unknown flag: $1"; exit 1 ;;
     esac
 done
 
-# Save full run output under /tmp/ptp-operator (timestamped file; also shown on the terminal).
-mkdir -p /tmp/ptp-operator
-RUN_ON_VM_LOG="/tmp/ptp-operator/run-on-vm-$(date +%Y%m%d-%H%M%S).log"
+# Per-run temp directory shared by all child scripts.
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+export PTP_RUN_DIR="${REPO_ROOT}/.local-runs/$(date +%Y%m%d-%H%M%S)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)